scheduler/server/job_state.go

```diff
@@ -22,10 +24,13 @@ type jobState struct {
 	TimeMarker     time.Time    //when was this job last marked (i.e. for reporting purposes)
 
 	// track tasks by state (completed, running, not started) for priority based scheduling algorithm
-	// TODO remove if we don't use the priority based scheduling algorithm
+	// Completed and Running are only used by priority based scheduling algorithm
 	Completed  map[string]*taskState
 	Running    map[string]*taskState
 	NotStarted map[string]*taskState
+
+	jobClass                       string
+	tasksByJobClassAndStartTimeSec map[string]map[time.Time]map[string]*taskState
```

Comment:
is there any way we can simplify this field? the nesting of the triple map is confusing imo

Reply:
converting to nested structs - hopefully that is better